### PR TITLE
Feature/azure blob storage for profile store

### PR DESCRIPTION
**Pull request by hassansuhaib**
Issue: https://github.com/livecycle/preevy/issues/102

Implemented Azure Blob storage for Profile Store

This approach requires these environment variables to be set by the user: `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET` because we can't use other methods for creating a blob service account as they require different keys and we only get URL as a param.

### CODE REVIEW

File: packages/driver-azure/src/fs/index.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    },
+    delete: async (filename: string) => {
+      const blockBlobClient = containerClient.getBlockBlobClient(filename)
+      const options: BlobDeleteOptions = {
+        deleteSnapshots: 'include',
+      }
+      const blobDeleteIfExistsResponse: BlobDeleteIfExistsResponse = await blockBlobClient.deleteIfExists(options)
+
+      if (blobDeleteIfExistsResponse.errorCode) {
+        throw new Error(`Error: ${blobDeleteIfExistsResponse.errorCode}`)
```
Comment (royra) [inline packages/driver-azure/src/fs/index.ts:85]: according to [this](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/storage/storage-blob/test/blobclient.spec.ts#L344), the `errorCode` will be set if the blob is not found - the contract requires to not throw.

File: packages/driver-azure/src/fs/index.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      const blobClient = containerClient.getBlobClient(filename)
+      const result = await blobClient.download()
+      if (result.readableStreamBody !== undefined) {
+        return await streamToBuffer(result.readableStreamBody)
+      }
+      return undefined
+    },
+    write: async (filename: string, content: Buffer | string) => {
+      const blockBlobClient = containerClient.getBlockBlobClient(filename)
+      await blockBlobClient.upload(content, content.length)
```
Comment (royra) [inline packages/driver-azure/src/fs/index.ts:75]: what happens if the uploaded blob already exists? should overwrite.

**Review by royra — CHANGES_REQUESTED** [review]
Thanks for your contribution!

- Some comments below to address
- Make sure the code is tested manually, or add automated tests
- Add the option to store the profile on Azure in the [`init`](https://github.com/livecycle/preevy/blob/42711d715/packages/cli/src/commands/init.ts#L91) command
- Add to the docs

File: packages/cli/src/fs.ts [hunk 3]
```diff
@@ -17,5 +18,10 @@ export const fsFromUrl = async (url: string, localBaseDir: string) => {
     // eslint-disable-next-line @typescript-eslint/return-await
     return await googleCloudStorageFs(url)
   }
+  if (fsType === 'az') {
```
Comment (Yshayy) [inline packages/cli/src/fs.ts:21]: Thanks for the contribution!

The type of the fsType as well as the URL should use `azblob` protocol and be conforming to this format:
https://gocloud.dev/howto/blob/#azure

#102

Comment (hassansuhaib) [reply packages/cli/src/fs.ts:21]: I saw this link in the issue but I got confused as all of the Azure documentation was using `https://`. I guess I'll have to understand the whole architecture of Preevy deeply in order to figure this out.

**Comment by royra** [comment]
fixed by #423

**Pull request closed without merging**